server/py/services/alerts/main.py

```diff
@@ -451,6 +451,48 @@ async def _filter_alert_activations_by_permissions(_alert_activations):
             pagination=page_info,
         )
 
+    async def get_alert_activation(
+        self,
+        request: fastapi.Request,
+        project: str,
+        name: Optional[str],
+        activation_id: int,
+        auth_info: mlrun.common.schemas.AuthInfo,
+        db_session: sqlalchemy.orm.Session = None,
+    ) -> mlrun.common.schemas.AlertActivation:
+        # if alert name is given, permission check can be done now,
+        # if not, it will be done after getting activation
+        if name:
+            await framework.utils.auth.verifier.AuthVerifier().query_project_resource_permissions(
+                mlrun.common.schemas.AuthorizationResourceTypes.alert_activations,
+                project,
+                name,
+                mlrun.common.schemas.AuthorizationAction.store,
+                auth_info,
+            )
+
+        alert_activation = await run_in_threadpool(
+            services.alerts.crud.AlertActivation().get_alert_activation,
+            db_session,
+            activation_id,
+        )
+        if not name:
+            await framework.utils.auth.verifier.AuthVerifier().query_project_resource_permissions(
+                mlrun.common.schemas.AuthorizationResourceTypes.alert_activations,
+                project,
+                alert_activation.name,
+                mlrun.common.schemas.AuthorizationAction.store,
+                auth_info,
+            )
+        else:
+            if alert_activation.name != name:
+                raise mlrun.errors.MLRunNotFoundError(
+                    f"Alert activation not found. "
+                    f"activation_id={activation_id}, "
+                    f"name={name}"
+                )
+        return alert_activation
```

Comment:
flow is a bit complicated, Id change it to
1. get the alert activiation
2. permission check 
3. ensure input name and alert activiation name is align
4. return activiation